src/CQRS/LeanCode.CQRS.AspNetCore/Local/LocalHttpContext.cs

```diff
@@ -0,0 +1,63 @@
+using System.Security.Claims;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace LeanCode.CQRS.AspNetCore.Local;
+
+internal class LocalHttpContext : HttpContext
+{
+    private readonly HttpContext inner;
+    private readonly LocalFeatureCollection features;
+
+    private IServiceProvider requestServices;
+    private CancellationToken requestAborted;
+
+    public LocalHttpContext(HttpContext inner, IServiceProvider requestServices, CancellationToken requestAborted)
+    {
+        this.inner = inner;
+        this.requestServices = requestServices;
+        this.requestAborted = requestAborted;
+
+        features = new LocalFeatureCollection(inner.Features);
+    }
+
+    public override HttpResponse Response =>
```

Comment:
My point here is that _some_ middlewares / handlers may want to use it and don't know that they're in a local context. It's better to let them write to /dev/null then raise an exception IMO.

Unless you want to be explicit with a local pipeline and make the end user responsible for enforcing it.

Reply:
The problem with that is _how_ should that work - we're not in real HTTP context. We can pretend that there is a request because it is read-only, but the same thing for response does not work at all.

The more I think about it, the more I am in favor of explicit local pipeline, where the user is required to define both pipelines separately.

Reply:
You cannot just instantiate `HttpResponse` (`DefaultHttpResponse` ?) and have it completely in memory? Maybe a custom null object implementation?

Reply:
Done